src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareIncludeExcludeNodeOperation.cs

```diff
@@ -66,27 +66,13 @@ public void Execute(TaskExecutionMode mode)
                     throw new InvalidOperationException(SR.SchemaCompareExcludeIncludeNodeNotFound);
                 }
 
-                // Check first if the dependencies will allow this if it's an exclude request
-                if (!this.Parameters.IncludeRequest)
-                {
-                    IEnumerable<SchemaDifference> dependencies = this.ComparisonResult.GetExcludeDependencies(node);
-
-                    bool block = dependencies.Any(d => d.Included);
-                    if (block)
-                    {
-                        this.Success = false;
-                        return;
-                    }
-                }
-
                 this.Success = this.Parameters.IncludeRequest ? this.ComparisonResult.Include(node) : this.ComparisonResult.Exclude(node);
 
-                // create list of affected dependencies of this request
-                if (this.Success)
-                {
-                    IEnumerable<SchemaDifference> dependencies = this.ComparisonResult.GetIncludeDependencies(node);
-                    this.ChangedDifferences = dependencies.Select(difference => SchemaCompareUtils.CreateDiffEntry(difference, null)).ToList();
-                }
+                // create list of dependencies of this schema difference
+                // if not successful, send exclude dependencies that caused it to fail
+                // if successful, send dependencies that might have been affected by this request
+                IEnumerable<SchemaDifference> dependencies = this.Success ? this.ComparisonResult.GetIncludeDependencies(node) : this.ComparisonResult.GetExcludeDependencies(node);
```

Comment:
not sure I understand this why are we deciding whether to get exclude dependencies or include based on success? technically we can have include operation that failed too - right? Am I missing something here?

Reply:
changed this in the next commit. If successful, sends back dependencies that might have been affected from tthe GetIncludedDependencies() call. If not successful, send back the blocking dependents which are from the GetExcludedDependencies() call.